code/modules/goals/goal_mind.dm

```diff
@@ -39,7 +41,6 @@
 		var/min_goals = 1
 		var/max_goals = 3
 		if(job && LAZYLEN(job.possible_goals))
-			available_goals |= job.possible_goals
```

Comment:
Wait, why move this?

Reply:
Because before it was only called if the add_amount was null.
This was only the case when the goals were generated at spawn, meaning that job-goals would only generate at spawn.
The move makes it so job-goals generate when clicking "reroll goal" or "generate new goal" as well